modules/RsIskraMeter/src/main.rs

```diff
@@ -0,0 +1,1197 @@
+        let mut signature = to_hex_string(regs);
+        signature.truncate((length_of_signature * 2) as usize);
+        log::info!("Read the signature: {}", signature);
+        Ok(signature)
+    }
+
+    /// For 15 seconds and checks the signature status every 10 ms
+    fn check_signature_status(&self) -> Result<()> {
+        retry::retry(
+            || {
+                let status = self.read_command_status()?;
+                match status {
+                    SignatureStatus::NotInitialized
+                    | SignatureStatus::Idle
+                    | SignatureStatus::SignatureInProgress => Ok(retry::RetryState::KeepTrying),
+                    SignatureStatus::SignatureOK => Ok(retry::RetryState::Done),
+                    error => anyhow::bail!("Error state {error:?}"),
+                }
+            },
+            Duration::from_secs(15),
+        )
+    }
+
+    fn start_transaction(
+        &self,
+        req: generated::types::powermeter::TransactionReq,
+    ) -> Result<TransactionStartResponse> {
+        // We can start transaction only in Idle state
+        let state = self.read_state()?;
+        match state {
+            IskraMaterState::Active
+            | IskraMaterState::Active_after_power_failure
+            | IskraMaterState::Active_after_reset => {
+                log::error!("Unexpected state {state:?}, trying to stop transaction");
+                self.stop_transaction()?;
+            }
+            IskraMaterState::Idle => {}
+            IskraMaterState::Unknown => {
+                log::warn!("Unknown state");
+            }
+        }
+        self.set_time()?;
+        log::info!("Set time");
+        // set algorithm
+        self.write_single_register(7059, 0)?;
+        self.write_metadata(&req.evse_id)?;
+
+        // Finally send start transaction, we are sending 'B'
+        self.write_single_register(7051, 0x4200)?;
+        log::info!("Started transaction.");
+
+        // Wait until the meter is active.
+        retry::retry(
+            || {
+                let state = self.read_state()?;
+                match state {
+                    IskraMaterState::Active => Ok(retry::RetryState::Done),
+                    _ => Ok(retry::RetryState::KeepTrying),
+                }
+            },
+            std::time::Duration::from_secs(1),
+        )?;
+
+        self.check_signature_status()?;
+        let signature = self.read_signature()?;
+        let signed_meter_values = self.read_signed_meter_values()?;
+        Ok(TransactionStartResponse {
+            error: Option::None,
+            signed_meter_value: Some(SignedMeterValue {
+                signed_meter_data: create_ocmf(signed_meter_values, signature),
+                signing_method: "OCMF".to_string(),
+                encoding_method: String::new(),
+                public_key: self.read_public_key().ok(),
+                timestamp: None,
```

Comment:
@hikinggrass @golovasteek should we set also the tiemstamps?

Reply:
Generally it makes sense for consistency, but only if we decode the OCMF string and take it from there.